### PR TITLE
tests:dma and adc upgrade to new ztest api

### PR DESCRIPTION
**Pull request by hakehuang**
1. config ztest_new_api and update the testcase defines

2. remove shell interactive test cases, twister has another plan

3. add sg case support for kinetis and rt10xx platfrom

related: #47002

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

**Comment by hakehuang** [comment]
test  pass on frdm_k64f and mimxrt1060_evk

**Review by danieldegrasse — DISMISSED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/drivers/dma/loop_transfer/src/test_dma_loop.c [hunk 1]
```diff
@@ -344,13 +344,13 @@ static int test_loop_suspend_resume(void)
 
 
 /* export test cases */
-void test_dma_m2m_loop(void)
+ZTEST(dma_m2m_loop_test, dma_m2m_loop)
```
[the remark this answers is not in the document]
Comment (hakehuang) [reply tests/drivers/dma/loop_transfer/src/test_dma_loop.c]: I see, updated

**Comment by hakehuang** [comment]
@nashif  I remove the '_test' postfix and add 'test_' to test case name back.

**Comment by hakehuang** [comment]
@teburd can you help to review?

**Review by nashif — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**